com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -266,39 +265,41 @@ public void onError(Exception e) {
 
     }
 
-    protected Runnable getUploadAppStepRunnable(DelegateExecution context, CloudApplicationExtended app, CloudFoundryOperations client,
+    protected Runnable getUploadAppStepRunnable(ExecutionWrapper execution, CloudApplicationExtended app, CloudFoundryOperations client,
         ClientExtensions clientExtensions) {
         return () -> {
-            String processId = context.getProcessInstanceId();
+            String processId = execution.getContext().getProcessInstanceId();
             getStepLogger().trace("Started upload app step runnable for process \"{0}\"", processId);
-            ExecutionStatus status = ExecutionStatus.FAILED;
-            Map<String, Object> outputVariables = new HashMap<>();
+            AsyncExecutionState status = AsyncExecutionState.ERROR;
             try {
-                String appArchiveId = StepsUtil.getRequiredStringParameter(context, Constants.PARAM_APP_ARCHIVE_ID);
-                String fileName = StepsUtil.getModuleFileName(context, app.getModuleName());
+                String appArchiveId = StepsUtil.getRequiredStringParameter(execution.getContext(), Constants.PARAM_APP_ARCHIVE_ID);
+                String fileName = StepsUtil.getModuleFileName(execution.getContext(), app.getModuleName());
                 getStepLogger().debug("Uploading file \"{0}\" for application \"{1}\"", fileName, app.getName());
                 if (clientExtensions != null) {
-                    String uploadToken = asyncUploadFiles(context, clientExtensions, client, app, appArchiveId, fileName);
-                    outputVariables.put(Constants.VAR_UPLOAD_TOKEN, uploadToken);
+                    String uploadToken = asyncUploadFiles(execution.getContext(), clientExtensions, client, app, appArchiveId, fileName);
+                    execution.getContextExtensionDao().addOrUpdate(execution.getContext().getProcessInstanceId(),
+                        Constants.VAR_UPLOAD_TOKEN, uploadToken);
                     getStepLogger().debug("Started async upload of application \"{0}\"", fileName, app.getName());
-                    status = ExecutionStatus.RUNNING;
+                    status = AsyncExecutionState.RUNNING;
                 } else {
-                    uploadFiles(context, client, app, appArchiveId, fileName);
-                    getStepLogger().debug(Messages.APP_UPLOADED, app.getName());
-                    status = ExecutionStatus.SUCCESS;
+                    uploadFiles(execution.getContext(), client, app, appArchiveId, fileName);
+                    getStepLogger().info(Messages.APP_UPLOADED, app.getName());

```

Comment:
Was this changed for debugging purposes?

Reply:
done